### DISCUSSION TITLE
Need help in debugging the service worker

### DISCUSSION THREAD

**Discussion by adyanth** [giscus/giscus#163] [Q&A]
Hey @laymonage 

I wanted to port this to cloudflare workers, and was working towards that. Could you help me to debug the build which fails with no errors explicitly shown?

```
adyanth@ubuntu-nuc:~/workspace/giscus-static$ npm run build

> giscus@0.1.0 build
> webpack

asset worker.js 1.19 KiB [compared for emit] (name: main) 1 related asset
./api/discussions/index.ts 39 bytes [built] [code generated]

ERROR in ./api/discussions/index.ts
Module build failed (from ./node_modules/ts-loader/index.js):
Error: TypeScript emitted no output for /home/adyanth/workspace/giscus-static/api/discussions/index.ts.
    at makeSourceMapAndFinish (/home/adyanth/workspace/giscus-static/node_modules/ts-loader/dist/index.js:52:18)
    at successLoader (/home/adyanth/workspace/giscus-static/node_modules/ts-loader/dist/index.js:39:5)
    at Object.loader (/home/adyanth/workspace/giscus-static/node_modules/ts-loader/dist/index.js:22:5)

webpack 5.50.0 compiled with 1 error in 2846 ms
```

Here is the branch for the serverless functions: https://github.com/adyanth/giscus/tree/workers

TIA!

**Comment by laymonage** [comment]
Hey,

You'll probably need to edit the [tsconfig.json](https://github.com/adyanth/giscus/blob/workers/tsconfig.json#L13) file and remove this line:

```ts
    "noEmit": true,
```

That might not be enough, though. This [template](https://github.com/cloudflare/worker-typescript-template) might be useful for reference.

Also, I recently changed the widget page to use Next.js SSR so that we can validate the `giscus.json` config from server-side. The page might be broken if generated statically (with `next export`). It's likely that you'll have to change the page to use `router.query` again. See the [history](https://github.com/laymonage/giscus/commits/main/pages/widget.tsx) for details, especially starting from [this commit](https://github.com/laymonage/giscus/commit/1f27a44501cbe91f36383a79d79d92f4d4158fea).

**Reply by adyanth** [reply]
Thanks! Setting noEmit to false got me some issues with webpack polyfills for the JWA/JWS libraries. Will look into that. I did use the cloudflare templates to set this up. Had not seen the lack of noEmit there, my bad :)

Regarding SSR, I did see that my static site failed to compile after pulling the new commits due to it. But what I am thinking is, do I really need to host that static site myself? Can I not just generate the client.js and just host that file along with the serverless API?
